mlflow/utils/conda.py

```diff
@@ -60,6 +61,30 @@ def _get_conda_env_name(conda_env_path, env_id=None):
     return "mlflow-%s" % hashlib.sha1(conda_env_contents.encode("utf-8")).hexdigest()
 
 
+def _get_conda_executables():
```

Comment:
What's the reason for not using mamba for both operation and env creation? Allowing 2 conda executables could confuse most users without the context. I'd prefer to not doing so if possible.

Reply:
yeah, I know, but unfortunately `mamba` provides only a subset of conda commands, specifically install, create, list, search, run, info and clean. For example, the `init` command is not available.

Hence, we must use this awkward thing of having two executables. On the other hand, `mamba` is installed using `conda`, so in most cases you have both.

Reply:
I see.

I'm a little confused reading the function comment below. It appears Mamba's usage is mostly limited to only create environments (as defined by MLFLOW_CONDA_CREATE_ENV_CMD). Is it correct? Do you mind listing the value of MLFLOW_CONDA_CREATE_ENV_CMD and MLFLOW_CONDA_HOME in the proposed use case when mamba and conda are both involved?

If my understanding is correct, how about a slightly different approach (mostly to reduce confusion):
1. define MLFLOW_MAMBA_HOME instead of MLFLOW_CONDA_CREATE_ENV_CMD
2. add a function _get_mamba_bin_executable(), similar to get_conda_bin_executable() to return mamba if defined, otherwise return conda instead (by calling get_conda_bin_executable() internally probably).
3. In get_or_create_conda_env(), call _get_mamba_bin_executable() and you are done.

Reply:
Yes, the usage of `mamba` is only limited to creating environment. 

If you want to use ``conda`` for both creating and activating environments, you don't need to do anything. This works as it has always worked. If you don't set anything, then ``conda`` is used from your ``PATH``. If you set ``MLFLOW_CONDA_HOME``, then ``${MLFLOW_CONDA_HOME}/bin/conda`` is used.

If you want to use mamba instead of conda to create environments, you just set:

```bash
export MLFLOW_CONDA_CREATE_ENV_CMD=mamba
```
Everything else works the same. So if you do not set `MLFLOW_CONDA_HOME`, then `mamba` will be assumed to be in one of the directories within the `PATH` env variable. If you do set it, then `mamba` will be executed from `${MLFLOW_CONDA_HOME}/bin/mamba`.

One reason to do it this way is to be a little future proof. Let's imagine that a third implementation of the `conda` package manager comes out in the future (i.e., an alternative to both conda and mamba), let's imagine it is called `boa`. Then you just set `MLFLOW_CONDA_CREATE_ENV_CMD=boa` and you are done. In other words, my implementation does not necessarily assume that the alternative implementation of `conda` is specifically `mamba`.

What do you think? I am open to go your route, although that would restrict us explicitly to mamba only.

Reply:
OK. How about this? Instead of making _get_conda_executables() returns 2 executables, add a separate function called _get_conda_executable_for_create_env() to return a dedicated executable for creating environment. With that it is clear what that executable is for, and how it can be obtained.

The rest of the code and remain mostly unchanged.